mixer/adapter/stackdriver/metric/bufferedClient.go

```diff
@@ -165,8 +152,6 @@ func (b *buffered) Send() {
 		// We need to build framework level support for these kinds of async tasks. Perhaps a generic batching adapter
 		// can handle some of this complexity?
 		if err != nil {
-			ets := handleError(err, timeSeries)
```

Comment:
What about transport errors?Are even those not retried?

Do we need to do anything in v2 for this ?

Reply:
The monitoring client actually has some default retry logic (https://github.com/googleapis/google-cloud-go/blob/master/monitoring/apiv3/metric_client.go#L61), which does nothing for `CreateTimeseries` because it is not idempotent and I think there is no status code reliably indicating whether request reaches server or not. `Unavailable` seems to be the closest one, but it could still happen after server received request (https://github.com/grpc/grpc/blob/master/doc/statuscodes.md). Also if there is indeed some retry strategy that we want apply here, we should upstream it to the monitoring client. For v2 we could add whatever the monitoring client does to the opencensus.

@jkohen @quentinmit Should we retry CreateTimeSeries for some particular gRPC status code? Does stackdriver server deal with duplicated CreateTimeSeries request?

Reply:
You should retry transient failures. I believe in gRPC and with this API, that's mostly `Unavailable`. You don't want to retry others like `Invalid argument`, which should never succeed. While the API doesn't document all the errors it may return, I believe those two and `deadline exceeded` are the only ones at this time.

It makes sense to have a deadline after which the data is discarded. In some monitoring applications that's probably before the next collection interval; in others you want to write every single point, if possible, and deadline should be infinite.

The reason we removed the partial errors is that there was bad correlation between individual elements in the response and actual failures to write those points. We understand the potential for regression, but in many cases the errors were over reporting failures, and due to size constraints in some gRPC and HTTP implementations, could only return 2-3 payloads even if there were more failures. For those reasons, we expect the impact to be low and in some cases positive. Unfortunately, due to implementation details not something we could have easily solved on the backend. @nicktrav

Reply:
One more thing. I just remembered that due to a backend misconfiguration the detailed error was only returned to clients for ~1 week. Short after enabling that, it started causing the aforementioned issues for the existing clients and we decided to retire that from the response. In any case, sorry for the confusion!